### PR TITLE
Add OIDC auth role and functional test

### PR DESCRIPTION
**Pull request by Lynty**
The HVAC library currently does not support OIDC auth role configuration so I have used the Python requests library to implement this feature.

 - [x] test for idempotency
 - [x] create `ansible-modules-hashivault/functional/test_oidc_auth_role.yml`

### CODE REVIEW

File: ansible/modules/hashivault/hashivault_oidc_auth_role.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python
+from ansible.module_utils.hashivault import hashivault_argspec
+from ansible.module_utils.hashivault import hashivault_auth_client
+from ansible.module_utils.hashivault import hashivault_init
+from ansible.module_utils.hashivault import hashiwrapper
+import json
+from ast import literal_eval
```
Comment (drewmullen) [inline ansible/modules/hashivault/hashivault_oidc_auth_role.py]: this library is not used and can be removed

File: ansible/modules/hashivault/hashivault_oidc_auth_role.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+    # check if engine is enabled
+    try:
+      if (mount_point + "/") not in client.sys.list_auth_methods()['data'].keys():
+        return {'failed': True, 'msg': 'auth method is not enabled', 'rc': 1}
+    except:
+        if module.check_mode:
+            changed = True
+        else:
+            return {'failed': True, 'msg': 'auth mount is not enabled', 'rc': 1}
```
Comment (drewmullen) [inline ansible/modules/hashivault/hashivault_oidc_auth_role.py]: can you update the output to say: "auth mount is not enabled or namespace is not created"

File: ansible/modules/hashivault/hashivault_oidc_auth_role.py [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+    # check if role exists
+    try:
+        current_state = requests.get(url + '/v1/auth/' + mount_point + '/role' + '/gmail', verify=verify, headers=headers)
```
Comment (drewmullen) [inline ansible/modules/hashivault/hashivault_oidc_auth_role.py]: do we specifically want /gmail?

Comment (Lynty) [reply ansible/modules/hashivault/hashivault_oidc_auth_role.py]: good catch!

Made changes and tested 👍

**Comment by TerryHowe** [comment]
Thanks!

**Pull request merged**